controllers/topics/index.js

```diff
@@ -170,6 +170,52 @@ const getTopics = async (req, res) => {
   }
 };
 
+const createTopic = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+
+    const {token, userId} = req;
+
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    const topic_id = await insertTopics([name]);
+    await followTopicViaCreateTopic({token, name, topic_id, userId});
+
+    return res.status(200).json({
+      success: true,
+      message: 'Success',
+      topic_id
+    });
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `createTopic` function, you are converting the topic name to lowercase (line 176). This could potentially lead to unexpected behavior if topic names are case-sensitive elsewhere in your application. If topic names are not intended to be case-sensitive throughout your application, this is fine. However, if they are, you should remove this line.

Also, error handling could be improved. Currently, you're returning a JSON response with an error message directly from the catch block (lines 204-209). It would be better to throw the error and let a global error handler take care of sending the appropriate HTTP response. This way, you can keep your controller functions clean and focused on their main responsibilities.

```diff
const createTopic = async (req, res, next) => {
  try {
    let {name} = req.body;
-    176:     name = name.toLowerCase();

    const {token, userId} = req;

    let topics = await Topics.findOne({
      where: {
        name,
        deleted_at: null
      }
    });

    if (topics) {
      return res.status(400).json({
        code: 400,
        status: 'failed',
        message: 'Topic already exists'
      });
    }

    const topic_id = await insertTopics([name]);
    await followTopicViaCreateTopic({token, name, topic_id, userId});

    return res.status(200).json({
      success: true,
      message: 'Success',
      topic_id
    });
  } catch (error) {
-    204:     return res.json({
-    205:       code: error.statusCode,
-    206:       status: 'fail',
-    207:       message: error.message,
-    208:       data: 'null'
-    209:     });
+    204:     next(error);
  }
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->